scripts/iam-tools/aws_creds_check.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+# Note: because we source this in the login shell, it needs to be compatible with bash and zsh.
+# There are several places where -n "$ZSH_NAME" is tested to see if we're running zsh, and then
+# the zsh version of the syntax is used instead of the bash version.
+
+# only bash v4+ and zsh are supported. do nothing for other shells
+if [ -n "$BASH_VERSION" -a "$BASH_VERSINFO" -gt 3 -o "$ZSH_NAME" ]; then
+
+function _check_creds()
+{
+    local -A accounts
+    local account_count creds_actual name num line regex now credfile=~/.aws/credentials expfile=~/.aws/credentials_expiration
```

Comment:
You're mixing variable assignments with variable scope declarations, which hampers readability.

I recommend altering this to split up the two tasks:

`local account_count creds_actual name num line <etc.>`
`local credfile=/some/file expfile=/other/file`